plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/hover/HoverProvider.java

```diff
@@ -76,31 +77,33 @@ public HoverProvider(EditorAgent editorAgent, TextDocumentServiceClient client,
         TextDocumentPositionParams paramsDTO = helper.createTDPP(document, context.getOffset());
 
         Promise<Hover> promise = client.hover(paramsDTO);
-        Promise<OrionHoverOverlay> then = promise.then(new Function<Hover, OrionHoverOverlay>() {
-            @Override
-            public OrionHoverOverlay apply(Hover arg) throws FunctionException {
-                OrionHoverOverlay hover = OrionHoverOverlay.create();
-                hover.setType("markdown");
-                String content = renderContent(arg);
-                // do not show hover with only white spaces
-                if (StringUtils.isNullOrWhitespace(content)) {
-                    return null;
-                }
-                hover.setContent(content);
-
-                return hover;
+        Promise<OrionHoverOverlay> then = promise.then((Function<Hover, OrionHoverOverlay>)arg -> {
```

Comment:
I think writing `(Hover arg)->....' would make the generic types clear and it feels nicer than a cast.